packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -55,9 +73,156 @@ export const ConnectedServices = () => {
+  // why they are disconnecting.
+  const [
+    confirmDisconnectModalRevealed,
+    revealConfirmDisconnectModal,
+    hideConfirmDisconnectModal,
+  ] = useBooleanState();
+  const [submitStartTime, setSubmitStartTime] = useState<number>();
+
+  // After the user confirms they want to disconnect from sync in Confirm Disconnect modal,
+  // if their reason was a lost/stolen device, or a suspicious device, then we show them
+  // an informative modal with some advice on next steps to take.
+  const [
+    adviceModalRevealed,
+    revealAdviceModal,
+    hideAdviceModal,
+  ] = useBooleanState();
+
+  const clearDisconnectingState = (
+    errorMessage?: string,
+    error?: ApolloError
+  ) => {
+    hideConfirmDisconnectModal();
+    DS.client = null;
+    DS.reason = null;
+    setSubmitStartTime(0);
+    if (errorMessage) {
+      alertBar.error(errorMessage, error);
+    }
+  };
+
+  const onConfirmDisconnect = (evt?: MouseEvent) => {
+    setSubmitStartTime(Date.now());
+
+    if (evt) {
+      // TODO: disabled ts-checking because the EventTarget type is missing
+      // 'closest'. Figure out how to most nicely tidy this up, even if that
+      // entails trying again with managed components / react state.
+      const modalEl = evt.target.closest('#modal');
+      const selected = modalEl.querySelector('input[type="radio"]:checked');
+      if (selected) {
+        DS.reason = selected.value;
+      }
+    }
+
+    if (!DS.client) {
+      return clearDisconnectingState('Client not found, unable to disconnect');
+    }
+
+    logViewEvent('settings.clients.disconnect', `submit.${DS.reason}`);
+
+    deleteConnectedService({
+      variables: {
+        input: {
+          clientId: DS.client.clientId,
+          deviceId: DS.client.deviceId,
+          sessionTokenId: DS.client.sessionTokenId,
+          refreshTokenId: DS.client.refreshTokenId,
+        },
+      },
+    });
+  };
+
+  const onSignOutClick = (client: AttachedClient) => {
+    DS.client = client;
+    // If it's a sync client, we show the disconnect survey modal.
+    // Only sync clients have a deviceId.
+    if (client.deviceId) {
+      revealConfirmDisconnectModal();
+    } else {
+      onConfirmDisconnect();
+    }
+  };
+
+  const onCloseAdviceModal = () => {
+    clearDisconnectingState();
+    hideAdviceModal();
+  };
+
+  const ATTACHED_CLIENT_DISCONNECT_MUTATION = gql`
+    mutation attachedClientDisconnect($input: AttachedClientDisconnectInput!) {
+      attachedClientDisconnect(input: $input) {
+        clientMutationId
+      }
+    }
+  `;
+
+  const [deleteConnectedService] = useMutation(
+    ATTACHED_CLIENT_DISCONNECT_MUTATION,
+    {
+      onCompleted: () => {
+        // TODO: followup issue for ensuring flow events are logged correctly,
+        // including utilizing the MetricsContext.
+        logViewEvent(
+          viewName,
+          `flow.timing.clients.disconnect.${Date.now() - submitStartTime!}`
```

Comment:
I don't think the current set of metrics function is really set up to log `flow` events. ☹️  This call logs a `settings.clients.disconnect.flow.timing.clients.disconnect.676`, but we don't want that view name prefix.

Reply:
@irrationalagent or @davismtl  I cannot recall how and where the flow timing metrics are used.  Can you link us to an example?

Reply:
Yikes, good catch. I'll remove this metrics event and file a followup issue